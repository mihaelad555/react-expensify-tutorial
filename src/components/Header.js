import React from 'react';
import {NavLink} from 'react-router-dom';
const Header = () => (
    <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}> Expensify Dashboard </NavLink>
    <NavLink to="/create" activeClassName="is-active">Create expense dashboard </NavLink>
    <NavLink to="/help" activeClassName="is-active">This is Help Page</NavLink>
    </header>
);
export default Header;