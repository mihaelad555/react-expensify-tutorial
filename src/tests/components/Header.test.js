import React from 'react'
import {shallow} from 'enzyme';
//import toJSON from 'enzyme-to-json';
//import ReactShallowRender from 'react-test-renderer/shallow'
import Header from '../../components/Header';
//react-test-render

test('should render Header', () => {
    const wrapper =shallow(<Header/>);
   // expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();
   // expect(wrapper.find('h1').length).toBe(1);
    //expect(wrapper.find('h1').text()).toBe('Expensify');
    // const renderer = new ReactShallowRender();
    // renderer.render(<Header/>);
    // console.log(renderer.getRenderOutput());
    // expect(renderer.getRenderOutput()).toMatchSnapshot();

    
})

