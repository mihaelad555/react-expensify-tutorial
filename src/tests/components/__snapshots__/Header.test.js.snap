// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
     Expensify Dashboard 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/create"
  >
    Create expense dashboard 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/help"
  >
    This is Help Page
  </NavLink>
</header>
`;
